crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -245,7 +237,18 @@ impl ServerBuilder {
 
     pub fn route(mut self, path: &str, method_handler: MethodRouter) -> Self {
         self.init_router();
-        self.router = self.router.map(|router| router.route(path, method_handler));
+        self.router = self.router.map(|router| {
```

Comment:
Looking at the documentation of [`route_layer`](https://docs.rs/axum/latest/axum/routing/struct.Router.html#method.route_layer), I don't think this is correct.

The `route_layer` function applies a new layer to all existing routes in the `Router`, which means every time we call this function to add a `route`, it's going to add a new instance of the middleware to all the existing routes -- not what we want, because we call `route` 4 times on `server_builder`, below.

Instead, I think the right thing to do is to make the calls to `route_layer` once, somewhere during `build`? Can you fix forward and update the cherry picks accordingly?

Reply:
Ah yes, this should fix it
https://github.com/MystenLabs/sui/pull/17477